rir/src/runtime/Code.h

```diff
@@ -93,18 +94,20 @@ struct Code : public RirRuntimeObject<Code, CODE_MAGIC> {
         }
         memcpy(&lazyCodeHandle_, h.c_str(), l);
         lazyCodeHandle_[MAX_CODE_HANDLE_LENGTH - 1] = '\0';
+        pending_ = false;
     }
     NativeCode nativeCode() {
         if (nativeCode_)
             return nativeCode_;
-        if (*lazyCodeHandle_ == '\0')
+        if (pending_ || *lazyCodeHandle_ == '\0')
             return nullptr;
         return lazyCompile();
     }
 
     bool isCompiled() const {
         return *lazyCodeHandle_ != '\0' && nativeCode_ != nullptr;
     }
+    bool pending() const { return pending_; }
```

Comment:
pending is a bit non-descript. can we make it more concrete? like `pendingCompilation` and maybe a comment about the possible in-between state?